packages/gatsby/src/internal-plugins/query-runner/query-runner.js

```diff
@@ -34,10 +34,15 @@ module.exports = async (pageOrLayout, component) => {
   if (result && result.errors) {
     report.log(
       report.stripIndent`
-        The GraphQL query from ${component.componentPath} failed
+        The GraphQL query from ${component.componentPath} failed.
+        See .cache/json/${pageOrLayout.jsonName} if develop mode.
 
         Errors:
           ${result.errors || []}
+        Path:
+          ${pageOrLayout.path || `unset`}
```

Comment:
`path` is empty for layouts, if this kind of information is valuable, layouts should show their `id`

Reply:
So `pageOrLayout.id` instead of `pageOrLayout.path || 'unset'`?

Reply:
I believe both have `component` set which is the file path to the component. The path here is the pathname in the browser.

Reply:
We output `component.componentPath` above that. I think reason to output path (for individual page) is to be able to pinpoint what node cause problems (but at this point, this would signal there might be problem in gatsby or plugin rather than in your site)? @docwhat I am correct or there is something else?

Reply:
ah right — ok, we should both a) only output the page pathname for pages and b) make it clear this is the page's path since file & browser paths are easily confused.

Reply:
The `path` is useful for finding problems in `gatsby-node.js`, which is where my problem was.